### PR TITLE
system-variables: add warning count

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

This adds basic docs for a mysql-compatible feature. It has been present in TiDB for a long time.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1221,6 +1221,12 @@ This variable is an alias for _transaction_isolation_.
 - Default value: 0
 - This variable controls the idle timeout of user sessions in seconds. A zero-value means unlimited.
 
+### warning_count
+
+- Scope: SESSION
+- Default value: 0
+- This read only variable indicates the number of warnings that occurred in the statement that was previously executed.
```
Comment (TomShawn) [inline system-variables.md:1228]:
```suggestion
- This read-only variable indicates the number of warnings that occurred in the statement that was previously executed.
```

**Review by TomShawn — APPROVED** [review]
LGTM
Any technical review needed?

**Comment by TomShawn** [comment]
@morgo Shall we only cherry-pick to v4.0 and v5.0? Earlier version docs have this note.
image

**Comment by dveeden** [comment]
/lgtm

However it *might* be good to add the range of possible values as well. See https://bugs.mysql.com/bug.php?id=82696 for a related issue in MySQL

**Review by dveeden — APPROVED** [review]

**Pull request merged**